src/styles/block/calendar.scss

```diff
@@ -0,0 +1,52 @@
+.calendar {
+  width: $calendar-width;
+  padding: $padding-size;
+
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-size;
+
+  @for $number from 28 to 31 {
+   &__month-length--#{$number} {
+      .calendar__day:nth-child(n + #{$number + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  @each $day, $indent in $days {
+    &__start-day--#{$day} {
+      .calendar__day:first-child {
+        margin-left: #{$indent};
+      }
+    }
+  }
+
+  &__day {
+    @include square($cell-size);
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    border: 1px solid coral;
+    border-radius: 10px;
+    background-color: $color;
+    cursor: pointer;
+    transition: 0.5s;
```

Comment:
Always add transition property. By default, transition property is all, and it's hard for the browser to control all properties